plugins/button/plugin.js

```diff
@@ -345,8 +345,8 @@
 					// For other buttons, do not remove the attribute, instead set its value (#2444).
 					element.setAttribute( 'aria-pressed', state === CKEDITOR.TRISTATE_ON );
 				} else if ( this.hasArrow ) {
-					// Indicates that menu button is opened (#421).
-					element.setAttribute( 'aria-expanded', state == CKEDITOR.TRISTATE_ON );
+					// Indicates that menu button is opened (#421, #5144).
+					element.setAttribute( 'aria-expanded', this._.on ? 'true' : 'false' );
```

Comment:
Currently, the `on` property is introduced in the `menubutton` plugin. The current hierarchy looks like this:
|--button
|---- menubutton <- defines and operates on internal `on` property

So now, in the lower hierarchy plugin, we are using the property defined in the higher hierarchy. This property means something in the `button` only if it is actually at least `menubutton` instance 🤔 So we have a button with an arrow, but we are unable to detect whether this button is expanded or not without a higher-level plugin (`menubutton`). That seems to be invalid in terms of proper inheritance.

What is more, the `panelbutton` plugin sets this property after invoking setState:
https://github.com/ckeditor/ckeditor4/blob/058f65bedfa674d166320dfed3e1f4fd17e430a8/plugins/panelbutton/plugin.js#L91-L93

This effectively means `colorbutoon` has `aria-expanded` always on false. The `language` button sets the `_.on` property on `1` and right after that on `0` **after** selecting language, which also makes this effectively that the `aria-expanded` is set on `false`.

Also, the aria-expanded does not change to `true` whether the menu actually opens. And even so, that is senseless for screen reader since the only way reader can read the state of the button is to leave focus (then menu is collapsed again) and focus it again (so it is folded at this moment).

Screen reader properly reads the attribute, but maybe... changing the aria attribute doesn't have a sense for such menus and always should be `false` or even removed? However, if it makes sense to update the `aria-attribute`:
- please correct the behaviour of `panelbutton`
- please correct the behaviour whenever the menubutton is expanded
- maybe we should declare `on` property in the `button` plugin: https://github.com/ckeditor/ckeditor4/blob/c08cfedb607fab87866052e4aff2ec8facfb115d/plugins/button/plugin.js#L93

Reply:
> Screen reader properly reads the attribute, but maybe... changing the aria attribute doesn't have a sense for such menus and always should be `false` or even removed?

The attribute should be added and reflect the current status of the menu associated with the button – that's how it's [described in the spec](https://w3c.github.io/aria/#aria-expanded). Let's do it properly.